src/openforms/appointments/utils.py

```diff
@@ -1,8 +1,86 @@
 
 import qrcode
 
+from openforms.appointments.base import (
+    AppointmentClient,
+    AppointmentLocation,
+    AppointmentProduct,
+)
+from openforms.appointments.exceptions import AppointmentCreateFailed
+from openforms.appointments.models import AppointmentsConfig
+from openforms.submissions.models import Submission
+
+
+def get_client():
+    config_path = AppointmentsConfig.get_solo().config_path
+    if not config_path:
+        raise ValueError("No config_path is specified in AppointmentsConfig")
+    config_class = import_string(config_path)
+    client = config_class.get_solo().get_client()
+    return client
+
+
+def book_appointment_for_submission(submission: Submission) -> None:
+    appointment_data = submission.get_merged_appointment_data()
+    expected_information = [
+        "productID",
+        "locationID",
+        "appStartTime",
+        "clientLastName",
+        "clientDateOfBirth",
+    ]
+
+    missing_appointment_information = []
+    not_filled_in_appointment_information = []
+    for key in expected_information:
+        if key not in appointment_data:
+            missing_appointment_information.append(key)
+        elif not appointment_data.get(key):
+            # Key was in form but not filled in
+            not_filled_in_appointment_information.append(key)
+
+    if len(missing_appointment_information) == len(expected_information):
+        # Submission was never intended to make an appointment so just return
+        return
+    elif missing_appointment_information or not_filled_in_appointment_information:
+        # Incomplete information to make an appointment
+        if missing_appointment_information:
+            submission.appointment_information += (
+                f"Missing information in form: "
+                f"{', '.join(missing_appointment_information)}. "
+            )
+        if not_filled_in_appointment_information:
+            submission.appointment_information += (
+                f"Information not filled in by user: "
+                f"{', '.join(not_filled_in_appointment_information)}. "
+            )
+        submission.save()
+        return
+
+    product = AppointmentProduct(identifier=str(appointment_data["productID"]), name="")
+    location = AppointmentLocation(identifier=appointment_data["locationID"], name="")
+    appointment_client = AppointmentClient(
+        last_name=appointment_data["clientLastName"],
+        birthdate=appointment_data["clientDateOfBirth"],
+    )
+    start_at = datetime.strptime(appointment_data["appStartTime"], "%Y-%m-%dT%H:%M:%S")
+
+    try:
+        client = get_client()
+        appointment_id = client.create_appointment(
+            [product], location, start_at, appointment_client
+        )
+        submission.appointment_information = f"Appointment Id: {appointment_id}"
+    except AppointmentCreateFailed as e:
+        submission.appointment_information = "Failed to make appointment"
+    finally:
+        submission.save()
```

Comment:
I did think about this and maybe this makes more sense but it also feels a bit like overkill.

Either `appointment_id` or `error_information` will be filled in and not both.  Also if there is an `appointment_id` we know it's successful and if there isn't then we know it's a failure so the `status` wouldn't really be needed so you'd really just have a model with two fields but only one or the other would be filled in.  Which is why I felt it may be overkill and we could just put the relevant info into a single field.

I am fine though if you think a one to one model is still a better solution but I'm curious what your thoughts are after reading the above?